radix-engine-interface/src/model/resource/non_fungible_id.rs

```diff
@@ -9,25 +9,88 @@ use sbor::*;
 use crate::abi::*;
 use crate::data::*;
 use crate::scrypto_type;
+use crate::math::Decimal;
+use crate::Describe;
 
 /// Represents a key for a non-fungible resource
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct NonFungibleId(pub Vec<u8>);
+pub struct NonFungibleId{ 
+    id_type: NonFungibleIdType,
+    value: Vec<u8>
```

Comment:
Would it make more sense fo have `NonFungibleId` as an `enum` and the value being the correct type e.g.:
```
pub enum NonFungibleId {
  String(String),
  Bytes(Vec<u8>),
  ..
}
```

Reply:
Reimplemented to `enum`.